### DISCUSSION TITLE
Breaking changes in version 4

### DISCUSSION THREAD

**Discussion by n3sted0wl** [ujjwalguptaofficial/JsStore#214] [Q&A]
# Title 
JsStore.Connection() -> initDb() silently failing to execute in version 4

# Description 
Previously working code started failing over the weekend after version 4.0.2 was published.
Execution fails to get past the indicated line:
![image](https://user-images.githubusercontent.com/12645462/118501704-014e0600-b6f7-11eb-9c0a-f15b894d02ab.png)

Code snippet:
initialize: async () => {
                    console.log("initializing PUMC.Persistence");
                    let getDbSchema = () => {
                        return {
                            name: "pumc_db",
                            tables: [
                                {
                                    version: 34,
                                    name: "request_logs",
                                    columns : {
                                        "PK"            : { primaryKey: true, autoIncrement: true },
                                        "name"          : { notNull: true, dataType: "string"     },
                                        "timestamp"     : { notNull: true, dataType: "date_time"  },
                                        "wasSuccessful" : { notNull: true, dataType: "boolean"    },
                                    }
                                },
                                {
                                    version : 34,
                                    name    : "last_server_responses",
                                    columns : {
                                        "name" : { notNull: true, dataType: "string" },
                                        "data" : { notNull: true, dataType: "array" },
                                    }
                                },
                            ]
                        };
                    };

                    dbConnection = new JsStore.Connection(new Worker("/assets/scripts/jsstore.worker.js"));
                    let newDb = await dbConnection.initDb(getDbSchema());
                    dbConnection.getDbVersion("pumc_db").then(function (version) {
                        if (newDb) {
                            console.log("Created DB pumc_db (v " + version + ")");
                        } else {
                            console.log("Opened DB pumc_db (v " + version + ")");
                        }
                    })
                },

**Comment by ujjwalguptaofficial** [comment]
`getDbVersion` has been removed in v4. Also please have a look at upgrading from v3 to v4 - https://jsstore.net/tutorial/v3-to-v4/
